EmployeeManagement.WebApi/Controllers/EmployeeController.cs

```
@@ -0,0 +1,145 @@
+            };
+
+            return Json(response, HttpStatusCode.OK);
+        }
+
+        /// <summary>
+        /// Get employees details
+        /// </summary>
+        /// <response code="200">Displayed Employees detail Success Response</response>
+        [HttpPost("EmployeeByIds")]
+        [ProducesResponseType(typeof(GetEmployeeRespose), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(void), (int)HttpStatusCode.NotImplemented)]
+        public async Task<IActionResult> GetEmployeesById(List<int> employeeIds)
+        {
+            IEnumerable<EmployeeModel> employees = await _employeeDomainService.GetEmployeesById(employeeIds);
+            GetEmployeeRespose response = new()
+            {
+                employeeResponse = _mappingCoordinator.Map<EmployeeModel, GetEmployeeResponseObject>((employees.ToList())).ToList()
+            };
+
+            return Json(response, HttpStatusCode.OK);
+        }
+
+        /// <summary>
+        /// Get employees details
+        /// </summary>
+        /// <response code="200">Displayed Employees detail Success Response</response>
+        /// <response code="404">Employees not found Response</response>
+        [HttpPut("EditEmployee")]
+        [ProducesResponseType(typeof(GetEmployeeRespose), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(string),(int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> EditEmployee(IEnumerable<EditEmployeeRequest> employee)
+        {
+            IEnumerable<EditEmployeeRequestModel> employeeToBeEdited = _mappingCoordinator.Map<EditEmployeeRequest, EditEmployeeRequestModel>(employee);
+            IEnumerable<EmployeeModel> employees = await _employeeDomainService.EditEmployee(employeeToBeEdited);
+            if (employees.Count() == 0)
+            {
+                return Json("Employee not found",HttpStatusCode.NotFound);
+            }
```

@Hari-Ram-Karthik This logic should not be a part of the controller layer.

---

Yes Puneet is right, Domain layer should have all the business logic. You can throw an exception from the domain and have a try-catch here to give the proper response.